pkg/destroy/destroyer.go

```diff
@@ -20,49 +20,38 @@ type Destroyer interface {
 	Run() error
 }
 
-// NewDestroyer returns Destroyer based on `metadata.json` in `rootDir`.
+// Creator is an interface for creating platform-specific destroyers.
+type Creator func(level log.Level, metadata *types.ClusterMetadata) (Destroyer, error)
```

Comment:
nit: `destroy.Creator` sounds odd but fine :)

Reply:
> nit: destroy.Creator sounds odd but fine :)

How about `PlatformNew`?  I'd like to rename `NewDestroyer` to `New` as well (destroyer-ness is implied by the package), mind if I pick that up here?

Reply:
> mind if I pick that up here

go ahead.

Reply:
> How about PlatformNew
maybe just `destroy.NewFunc`

Reply:
> maybe just `destroy.NewFunc`

Done in e65f8f7 -> 70a37ae.